src/sentry/replays/consumers/recording_buffered.py

```diff
@@ -275,8 +312,27 @@ def process_message(buffer: RecordingBuffer, message: Message[KafkaPayload]) ->
 
 def commit_uploads(upload_events: list[UploadEvent]) -> None:
     with sentry_sdk.start_span(op="replays.consumer.recording.upload_segments"):
+        # This will run to completion taking potentially an infinite amount of time. However,
+        # that outcome is unlikely. In the event of an indefinite backlog the process can be
+        # restarted.
         with ThreadPoolExecutor(max_workers=len(upload_events)) as pool:
-            pool.map(_do_upload, upload_events)
+            futures = [pool.submit(_do_upload, upload) for upload in upload_events]
+
+    has_errors = False
+
+    # These futures should never fail unless there is a service-provider issue.
+    for error in filter(lambda n: n is not None, (fut.exception() for fut in futures)):
+        has_errors = True
+        sentry_sdk.capture_exception(error)
```

Comment:
why not `raise error` right here?

Reply:
Trying to capture all of the errors not just one of them.  I could raise the last error.